src/git.rs

```diff
@@ -99,6 +100,27 @@ impl GitHelper {
             .url()
             .map(|s| s.to_string()))
     }
+
+    pub(crate) fn is_relevant(&self, commit: &Commit, files: &Vec<PathBuf>) -> bool {
+        if files.is_empty() {
+            return true;
+        }
+
+        let tree = commit.tree().ok();
+        let parent_tree = commit.parent(0).ok().map(|item| item.tree().unwrap());
+        let diff = self.repo.diff_tree_to_tree(parent_tree.as_ref(), tree.as_ref(), None).unwrap();
+        let mut any_relevant = false;
+
+        diff.foreach(&mut |file, _progress| {
+            let filename = file.new_file().path().unwrap().to_str().unwrap();
+            any_relevant |= files.iter().any(|item| {
+                filename.starts_with(item.to_str().unwrap())
+            });
+            true
+        }, None,None, None).ok();
```

Comment:
lot's of unwraps here. But as you indicate you don't know rust, I am already impressed you could put this feature into the code base.

I did not check this code in details, but it seems to work indeed. 

Some unwraps can be avoided:

```suggestion
        let tree = commit.tree().ok();
        let parent_tree = commit.parent(0).and_then(|item| item.tree()).ok();
        let diff = self.repo.diff_tree_to_tree(parent_tree.as_ref(), tree.as_ref(), None).unwrap();
        let mut any_relevant = false;

        diff.foreach(&mut |file, _progress| {
            let filename = file.new_file().path().and_then(Path::to_str).unwrap_or_default();
            any_relevant |= files.iter().any(|item| {
                item.to_str().map(|item| filename.starts_with(item)).unwrap_or(false)
            });
            true
        }, None,None, None).ok();
```

Reply:
I've refactored the code to improve it.